### PR TITLE
[codex] Redesign command wait runtime

### PR DESCRIPTION
**Pull request by Ahoo-Wang**
## Summary

Redesigns the command wait runtime around explicit wait plans, unicast runtime handles, and stateful wait state machines.

- Replaces the old `WaitStrategy` runtime with immutable `WaitPlan` intent plus `WaitHandle` execution contracts.
- Splits last-result and stream-result handling into `WaitLastHandle` and `WaitStreamHandle`.
- Moves stage and chain completion logic into `StageWaitState` and `ChainWaitState`.
- Removes unused wait runtime helpers and dead predicate/reducer code.
- Updates framework docs, onboarding docs, diagrams, tests, benchmarks, and propagation/tracing/webflux integrations to use wait-plan terminology.

## Why

The old wait runtime mixed intent, signal reduction, registration, and result delivery behind strategy-style abstractions. That made stream-vs-last behavior harder to reason about and left reducer/predicate code paths that no longer matched the desired architecture.

The new model keeps `WaitPlan` as immutable intent, registers a single runtime `WaitHandle` per `waitCommandId`, and lets the handle own the mutable `WaitState` that decides accepted/completed signals.

## Validation

- `./gradlew check`
- `./gradlew :wow-core:test --tests 'me.ahoo.wow.command.wait.*' --tests 'me.ahoo.wow.command.wait.chain.*' --tests 'me.ahoo.wow.command.wait.stage.*' --tests 'me.ahoo.wow.command.DefaultCommandGatewayTest'`
- `./gradlew :wow-core:compileKotlin`
- `./gradlew :wow-tck:detekt :wow-tck:compileKotlin`
- `./node_modules/.bin/vitepress build docs` from `documentation/`
- `./gradlew :wow-benchmarks:benchmarkQuickE2E :wow-benchmarks:benchmarkQuickInfrastructureE2E :wow-benchmarks:generateBenchmarkReport :wow-benchmarks:generateInfrastructureBenchmarkReport :wow-benchmarks:generateQuickBenchmarkReport`
- `git diff --check`
- stale-term scan for removed wait strategy/reducer names

## Benchmark Notes

Quick Framework E2E and Quick Infrastructure E2E reports were regenerated on 2026-06-14 local time with Redis on `localhost:6379` and MongoDB on `localhost:27017` available. Quick results are directional local feedback; Full E2E remains the formal performance evidence source.

### CODE REVIEW

File: wow-core/src/main/kotlin/me/ahoo/wow/command/DefaultCommandGateway.kt [hunk 1]
```diff
@@ -178,108 +179,118 @@ class DefaultCommandGateway(
...
-                waitStrategy
-                    .waitingLast()
-                    .map { waitSignal ->
-                        waitSignal
-                            .toResult(command)
-                            .apply {
-                                if (!succeeded) {
-                                    throw CommandResultException(this)
+        Mono.defer {
+            val handle = waitCoordinator.createLast(waitPlan)
```
[the remark this answers is not in the document]
Comment (Ahoo-Wang) [reply wow-core/src/main/kotlin/me/ahoo/wow/command/DefaultCommandGateway.kt]: Fixed in `f84ab9463f`.

`sendAndWait` / `sendAndWaitStream` now validate the wait plan and run `check(command)` before reserving a wait handle. I also added a regression test covering an in-flight duplicate command so it returns `CommandResultException` with `DUPLICATE_REQUEST_ID` instead of leaking the coordinator registration conflict.

Verification: `./gradlew :wow-core:check`.

File: wow-core/src/main/kotlin/me/ahoo/wow/command/CommandGateway.kt [hunk 2]
```diff
@@ -81,14 +81,14 @@ interface CommandGateway : CommandBus {
      *
      * @param C the type of the command
      * @param command the command message to send
-     * @param waitStrategy the strategy defining the processing stages to monitor
+     * @param waitPlan the plan defining the processing stages to monitor
      * @return a Flux emitting CommandResult objects as the command progresses
      * @see CommandResult
-     * @see WaitStrategy
+     * @see WaitPlan
```
[the remark this answers is not in the document]
Comment (Ahoo-Wang) [reply wow-core/src/main/kotlin/me/ahoo/wow/command/CommandGateway.kt:87]: Keeping this as an intentional breaking change for this PR.

The wait redesign explicitly removes the old `WaitStrategy` runtime contract and replaces it with `WaitPlan` plus runtime `WaitHandle` implementations. Adding deprecated `WaitStrategy` overloads would keep the old strategy-centered API surface alive and conflict with the technical-debt cleanup goal of this refactor. The PR body and architecture document call this out as a deliberate public API break.
